shotover-proxy/src/transforms/redis/timestamp_tagging.rs

```diff
@@ -36,152 +33,165 @@ impl RedisTimestampTaggerConfig {
+
+                let mut timestamps: Vec<RedisFrame> = vec![];
+                let mut results: Vec<RedisFrame> = vec![];
+                for v_u in values {
+                    if let RedisFrame::Array(v) = v_u {
+                        if v.len() == 2 {
+                            let timestamp = v.pop().unwrap();
+                            let actual = v.pop().unwrap();
+                            timestamps.push(timestamp);
+                            results.push(actual);
                         }
-                        MessageValue::Integer(seconds as i64 - i, IntSize::I64)
                     }
-                    v => v,
+                }
+
+                todo!("ConsistentScatter isn't built to handle multiple timestamps yet: {timestamps:?} {results:?}",)
+            } else if values.len() == 2 {
+                let timestamp = values.pop().unwrap();
+                let frame = values.pop().unwrap();
+
+                let timestamp = match timestamp {
+                    RedisFrame::Integer(i) => i,
+                    _ => 0,
                 };
-                MessageValue::Document(BTreeMap::from([("timestamp".to_string(), processed_value)]))
-            });
-            values.pop()
+
+                let since_the_epoch = SystemTime::now()
+                    .duration_since(UNIX_EPOCH)
+                    .expect("Time went backwards");
+                let seconds_since_the_epoch = since_the_epoch.as_secs();
+                if seconds_since_the_epoch.leading_ones() > 1 {
+                    panic!("Cannot convert u64 to i64 without overflow");
+                }
+                let meta_timestamp = seconds_since_the_epoch as i64 - timestamp;
+                Some(ToWrite {
+                    meta_timestamp,
+                    frame,
+                })
+            } else {
+                None
+            }
         } else {
-            Some(MessageValue::List(values))
+            None
         };
+
+        if let Some(write) = to_write {
+            *redis_frame = write.frame;
+            message.meta_timestamp = Some(write.meta_timestamp);
+        }
     }
+    Ok(())
 }
 
 #[async_trait]
 impl Transform for RedisTimestampTagger {
     async fn transform<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
-        // TODO: This is wrong. We need to keep track of tagged_success per message
-        let mut tagged_success = true;
-        // TODO: This is wrong. We need more robust handling of transactions
-        let mut exec_block = false;
-
-        for message in message_wrapper.messages.iter_mut() {
-            message.generate_message_details_query();
-            if let MessageDetails::Query(ref mut qm) = message.details {
-                if let Some(a) = &qm.ast {
-                    if a.get_command() == *"EXEC" {
-                        exec_block = true;
+        let mut tagged_success = vec![true; message_wrapper.messages.len()];
```

Comment:
Solving this is great, can it go in a seperate PR though?
Already a lot going on here.

Reply:
Sure, I'll fix this PR so it's only changes from #445.

Reply:
To be clear it doesnt have to exactly match the code from #445, theres likely things I missed in #445, but anything that changes logic unrelated to the goal of the message details redesign should go in a separate PR.